.jenkins/pipelines/Azure/Linux/Jenkinsfile

```diff
@@ -18,21 +18,86 @@ AGENTS_LABELS = [
     "acc-ubuntu-18.04-vanilla": env.UBUNTU_VANILLA_1804_CUSTOM_LABEL ?: "vanilla-ubuntu-1804"
 ]
 
+def CmakeArgs(String build_type = "RelWithDebInfo", String code_coverage = "OFF", String debug_malloc = "ON", String lvi_args="", String cmake_args = "") {
+    def args = "-G Ninja -DCMAKE_INSTALL_PREFIX:PATH='/opt/openenclave' -DCPACK_GENERATOR=DEB -DCODE_COVERAGE=${code_coverage} -DUSE_DEBUG_MALLOC=${debug_malloc} -DCMAKE_BUILD_TYPE=${build_type} ${lvi_args} ${cmake_args} -Wdev"
+    return args
+}
+
+def TestCommand() {
```

Comment:
With the exception of CmakeArgs() the other functions `TestCommand(), InstallBuildCommand(), InstallReleaseCommand(), and TestSamplesCommand()` can be simplfied to be strings, which will save you a few lines to maintain as you won't need to assign the function to a variable to use in your shell script.

For example,

```
def TestCommand = """
    echo "Running Test Command"
    ctest --output-on-failure --timeout ${CTEST_TIMEOUT_SECONDS}
"""
```
which is then referenced like so
```
def ACCUpgradeTest(String label, String compiler, String version, List extra_cmake_args = [], List test_env = []) {
    stage("${label} Container ${version} RelWithDebInfo, extra_cmake_args: ${extra_cmake_args}") {
        node("${label}") {
            timeout(GLOBAL_TIMEOUT_MINUTES) {
                cleanWs()
                checkout scm
                def cmakeArgs = CmakeArgs("RelWithDebInfo","OFF","ON","-DLVI_MITIGATION_BINDIR=/usr/local/lvi-mitigation/bin",extra_cmake_args.join(' '))
                def task = """
                           \$InstallReleaseCommand
                           \$TestSamplesCommand
                           cmake ${WORKSPACE} ${cmakeArgs}
                           ninja -v
                           \$InstallBuildCommand
                           \$TestSamplesCommand
                           """
                withEnv(test_env) {
                    oe.Run(compiler, task)
                }
            }
        }
    }
}
```

Reply:
Sorry the above example is not correct. \$variable references an environment variable. The correct way to reference a groovy variable here is:
```
                def example = """
                           \"${InstallReleaseCommand}\"
                           """
```